extensions/amp-sticky-ad/1.0/amp-sticky-ad.js

```diff
@@ -112,6 +114,41 @@ class AmpStickyAd extends AMP.BaseElement {
     return Promise.resolve();
   }
 
+  /** @override */
+  upgradeCallback() {
+    if (!isExperimentOn(this.win, 'amp-sticky-ad-to-amp-ad')) {
+      return null;
+    }
+
+    const children = realChildElements(this.element);
+    userAssert(
+      children.length == 1 && children[0].tagName == 'AMP-AD',
+      'amp-sticky-ad must have a single amp-ad child'
+    );
+
+    const ad = children[0];
+    const enableConversion = Math.random() < 0.5;
```

Comment:
Do you not want to use the randomlySetBranch helper?

Reply:
Each page could only have one amp-sticky-ad, so I do not need to persist the branch state on the `window` object. Without that, `randomlySetBranch` is essentially doing the same thing as `Math.random`, and I feel it is clearer than declaring something like

```
{
          experimentId: ZINDEX_EXP,
          isTrafficEligible: () => true,
          branches: Object.values(ZINDEX_EXP_BRANCHES),
}
```